File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>
#include <cstring>
using namespace std;

bool solution(string s)
{
    bool answer = true;
    int cnt = 0;
    stack<char> st;
    for (int i = 0; i < s.length(); i++) {
        if (s[i] == '(') {
            st.push('(');
        } else {
            if (st.empty()) {
                return false;
            } else {
                st.pop();
            }
        }
    }
    if (!st.empty()) {
        return false;
    }
    return true;
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    cnt = 0
    s = list(s)
    while s:
        tmp = s.pop()
        if tmp == '(':
            if cnt > 0:
                cnt -= 1
            else:
                return False
        else:
            cnt += 1
    if cnt:
        return False
    return True